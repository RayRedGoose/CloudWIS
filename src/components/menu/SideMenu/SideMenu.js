import React from 'react'
import './SideMenu.scss'
import BackIcon from '../_icons/BackIcon/BackIcon'
import SearchForm from '../SearchForm/SearchForm'
import SearchHistory from '../SearchHistory/SearchHistory'

const SideMenu = (props) => {
  return (
    <main>
      <section className="menu-panel">
        <BackIcon toggleActiveState={props.toggleActiveState} />
        <SearchForm
          changeCity={props.changeCity}
          fetchWeatherData={props.fetchWeatherData}
          toggleActiveState={props.toggleActiveState} />
        <SearchHistory {...props} />
      </section>
      <section className="darkscreen"></section>
    </main>
  )
}

export default SideMenu
