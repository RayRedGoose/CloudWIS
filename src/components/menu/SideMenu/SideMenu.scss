.menu-panel {
  background-color: #0f1a20;
  color: #fff;
  width: 35vw; height: 100%;
  padding: 20px;
  position: relative;
}

.darkscreen {
  background-color: #fff;
  opacity: 0.8;
  width: 100%; height: 100%;
}
