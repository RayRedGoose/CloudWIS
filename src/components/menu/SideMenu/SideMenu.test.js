import React from 'react';
import { shallow } from 'enzyme';
import SideMenu from './SideMenu';

it('should match the snapshot with all data passed in correctly', () => {
  const mockProps = {
    changeCity: jest.fn(),
    searchHistory: ['Denver', 'Phoenix'],
    fetchWeatherData: jest.fn(),
    toggleActiveState: jest.fn()
  }

  const menu = shallow(<SideMenu { ...mockProps } />);

  expect(menu).toMatchSnapshot();
});
