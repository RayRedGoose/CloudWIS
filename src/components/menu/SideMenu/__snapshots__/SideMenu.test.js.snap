// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<main>
  <section
    className="menu-panel"
  >
    <BackIcon
      toggleActiveState={[MockFunction]}
    />
    <SearchForm
      changeCity={[MockFunction]}
      fetchWeatherData={[MockFunction]}
      toggleActiveState={[MockFunction]}
    />
    <SearchHistory
      changeCity={[MockFunction]}
      fetchWeatherData={[MockFunction]}
      searchHistory={
        Array [
          "Denver",
          "Phoenix",
        ]
      }
      toggleActiveState={[MockFunction]}
    />
  </section>
  <section
    className="darkscreen"
  />
</main>
`;
